waspc/src/Wasp/Generator/WebAppGenerator/OperationsGenerator.hs

```diff
@@ -57,20 +65,54 @@ genActions spec =
 genQuery :: AppSpec -> (String, AS.Query.Query) -> Generator FileDraft
 genQuery _ (queryName, query) = return $ C.mkTmplFdWithDstAndData tmplFile dstFile (Just tmplData)
   where
-    tmplFile = C.asTmplFile [relfile|src/queries/_query.js|]
+    tmplFile = C.asTmplFile [relfile|src/queries/_query.ts|]
 
     dstFile = C.asWebAppFile $ [reldir|src/queries/|] </> fromJust (getOperationDstFileName operation)
     tmplData =
-      object
+      object $
         [ "queryRoute"
             .= ( ServerGenerator.operationsRouteInRootRouter
                    ++ "/"
                    ++ ServerOperationsRoutesG.operationRouteInOperationsRouter operation
                ),
           "entitiesArray" .= makeJsArrayOfEntityNames operation
         ]
+          ++ operationTypeData operation
     operation = AS.Operation.QueryOp queryName query
 
+operationTypeData :: AS.Operation.Operation -> [Pair]
```

Comment:
I'm going to reconsider where to put this entire function (and make it nicer too).